src/main/java/org/jabref/gui/linkedfile/LinkedFileDialogController.java

```diff
@@ -27,26 +30,46 @@ public class LinkedFileEditDialogView extends BaseDialog<LinkedFile> {
 
     @Inject private DialogService dialogService;
     @Inject private StateManager stateManager;
-
     @Inject private PreferencesService preferences;
 
     private LinkedFilesEditDialogViewModel viewModel;
-
     private final LinkedFile linkedFile;
 
-    public LinkedFileEditDialogView(LinkedFile linkedFile) {
+    /**
+     * Constructor for adding a new LinkedFile.
+     */
+    public LinkedFileDialogController() {
+        this(new LinkedFile("", "", ""));
+        initializeDialog(Localization.lang("Add file link"), ADD_BUTTON);
+    }
+
+    /**
+     * Constructor for editing an existing LinkedFile.
+     *
+     * @param linkedFile The linked file to be edited.
+     */
+    public LinkedFileDialogController(LinkedFile linkedFile) {
         this.linkedFile = linkedFile;
+        initializeDialog(Localization.lang("Edit file link"), EDIT_BUTTON);
+    }
 
+    /**
+     * Initializes the dialog with the given title and button type.
+     *
+     * @param title The title of the dialog.
+     * @param primaryButtonType The primary button type for the dialog.
+     */
+    private void initializeDialog(String title, ButtonType primaryButtonType) {
         ViewLoader.view(this)
                   .load()
                   .setAsContent(this.getDialogPane());
 
-        this.getDialogPane().getButtonTypes().addAll(ButtonType.APPLY, ButtonType.CANCEL);
+        this.setTitle(title);
         this.setResizable(false);
-        this.setTitle(Localization.lang("Edit file link"));
```

Comment:
Obsolete localization key. Refer to point 5 of the FAQs.
https://devdocs.jabref.org/code-howtos/faq.html

Reply:
No, this is used in line 53 (see above). And it is good that it is still used: The old dialog should keep the functionality.